package-lock.json

```diff
@@ -0,0 +1,4009 @@
+{
```

Comment:
To recap / confirm the recommended practice, it is desirable to add package-lock files to libraries (such as resin-bundle-resolve) but not to "apps" installed by users (such as balena-cli); is this right?

Reply:
But there was a good reason why `balena-cli` does not commit `package-lock.json` (it is in `.gitignore`), while the libs imported by the CLI can/should have `package-lock.json` files. @thgreasi, you might remember what the reason was... I think it was to do with we not having control over the customer's environment -- like, `package-lock.json` was _ignored_ (?) in some cases or platforms (?), so we wanted the CI to try installing the latest versions of whatever is in `package.json` -- but I am not able to fully justify it to myself. :-)

Reply:
I believe @Page- also has some theories on this too.